DDG4/hepmc/HepMC3EventReader.cpp

```diff
@@ -130,12 +130,24 @@ HEPMC3EventReader::readParticles(int event_number, Vertices& vertices, Particles
     p->genStatus = genStatus&G4PARTICLE_GEN_STATUS_MASK;
 
     if ( p->parents.size() == 0 )  {
+      // This is a bit unintuitive. A particle without a parent in HepMC
+      // can only be (something like) a beam particle, and it is attached
+      // to the root vertex, by default (0,0,0) and equal for all parent-less particles.
+      // Therefore, p->vsx etc. is not suitable to cover situations with multiple
+      // particles coming from different vertices.
+      // Intuitively, one should add vertices for all particles that have a parent,
+      // but heritage is handled differently in other places so the
+      // fix is to use these parentless particles' end vertices.
+      // Note that for a particle without end vertex (such as in a particle gun),
+      // it defaults to (0,0,0). This cannot be fixed, the information simply isn't in the HepMC file
+      // Having a parent enforces a vertex, having none forbids one.
```

Comment:
I don't understand where this phrase fits in?

Reply:
Since we're using end vertices, I looked at particles that don't have one and found the handle class defaults to 0. I played around trying to fix that but then realized there's nothing to save, the vertex information is simply gone. By all means shorten, change, clarify the comment, I just wrote what was going through my mind.